File: android/src/test/java/me/carda/awesome_notifications/awesome_notifications_core/utils/ListUtilsTest.java
package me.carda.awesome_notifications.awesome_notifications_core.utils;

import org.junit.Test;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.*;

public class ListUtilsTest {

    @Test
    public void isNullOrEmpty() {
        assertTrue("null values must are considered empty and must return true", ListUtils.isNullOrEmpty(null));
        assertTrue("null values must are considered empty and must return true", ListUtils.isNullOrEmpty(new ArrayList<>()));

        List test1 = new ArrayList<>();
        test1.add(1);
        assertFalse("List with values are considered not empty and must return false", ListUtils.isNullOrEmpty(test1));
    }

    @Test
    public void listHasDuplicates(){
        assertFalse("List with unique values do not have duplicates", ListUtils.listHasDuplicates(Arrays.asList(1,2,3,4,5,6)));
        assertTrue("List with not unique values have duplicates", ListUtils.listHasDuplicates(Arrays.asList(1,2,3,4,5,6,1)));
        assertTrue("List with unique values do not have duplicates", ListUtils.listHasDuplicates(Arrays.asList(1,1,2,3,4,5,6)));
        assertTrue("List with unique values do not have duplicates", ListUtils.listHasDuplicates(Arrays.asList(1,2,3,1,4,5,6)));
        assertTrue("List with unique values do not have duplicates", ListUtils.listHasDuplicates(Arrays.asList(1,1,1,1,1,1)));
    }
}

File: packages/awesome_notifications/test/src/utils/list_utils_test.dart
import 'package:awesome_notifications/src/utils/list_utils.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  group('isNullOrEmpty test', () {
    test('detect if list is null or empty', () {
      expect(AwesomeListUtils.isNullOrEmpty(null), true);
      expect(AwesomeListUtils.isNullOrEmpty([]), true);
    });

    test('detect if a non empty list is not empty', () {
      expect(AwesomeListUtils.isNullOrEmpty([null]), false);
      expect(AwesomeListUtils.isNullOrEmpty([1]), false);
      expect(AwesomeListUtils.isNullOrEmpty([1, 2]), false);
      expect(AwesomeListUtils.isNullOrEmpty([1, 2, 3]), false);
    });
  });
}
